primitiv/device_ops/cuda/max.cu

```diff
@@ -15,19 +15,12 @@ __global__ void max_fw_dev(
   px += bid % skip + (bid / skip) * skip * n;
   temp[tid] = FLOAT_NEGATIVE_INFINITY;
```

Comment:
Using a local variable, and substituting it to `temp` only once at the end of the loop.